### DISCUSSION TITLE
RedissonAutoConfiguration import not exist class with springboot 2.7.7

### DISCUSSION THREAD

**Discussion by donqing** [redisson/redisson#5607] [Q&A]
![image](https://github.com/redisson/redisson/assets/14136218/9c2077dd-ba1e-4a0a-bd01-6d053f630b14)
RedissonAutoConfiguration import "org.springframework.boot.autoconfigure.data.redis.RedisConnectionDetails" class which should not exist on classpath because my springboot version is 2.7.7. So as my understanding, RedissonAutoConfiguration shouldn't be compiled through,  so how does it auto-configured? 
redission-spring-boot-starter-3.26.0
pom eg:
 <dependency>
            <groupId>org.redisson</groupId>
            <artifactId>redisson-spring-boot-starter</artifactId>
            <exclusions>
                <exclusion>
                    <groupId>org.redisson</groupId>
                    <artifactId>redisson-spring-data-32</artifactId>
                </exclusion>
            </exclusions>
        </dependency>
        <dependency>
            <groupId>org.redisson</groupId>
            <artifactId>redisson-spring-data-27</artifactId>
        </dependency>

**Comment by mrniko** [comment]
It's compiled with the latest spring boot version.
